### PR TITLE
Separate attributes domain, subdomain and name in KuzzleError

### PR DESCRIPTION
**Pull request by Yoann-Abbes**
## What does this PR do ?

For now we received an `errorName` like `internal.service.invalid_thing` and we store it in the class.
So the user receveid `errorName = internal.service.error`
This PR separate in 3 attribute which is easier for users to manage.
Now he will receive 
```
domain = internal,
subdomain = service,
error = invalid_thing
```

### How should this be manually tested?

  - Step 1 :
  - Step 2 :
  - Step 3 :  
  ...

### CODE REVIEW

File: lib/errors/kuzzleError.js [hunk 1]
```diff
@@ -7,13 +7,16 @@ const util = require('util');
...
-  constructor(message, status, errorName = 'Undocumented error', code = 0) {
+  constructor(message, status, errorName = 'internal.errorsManager.undocumented_error', code = 0) {
     super(message);
 
     this.status = status;
 
     this.code = code;
-    this.errorName = errorName;
+    
+    this.domain = errorName.split('.')[0];
```
Comment (Aschen) [inline lib/errors/kuzzleError.js]: Use destructuring assignment

File: package.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
   "description": "Common objects shared to various Kuzzle components and plugins",
   "main": "./index.js",
   "author": "The Kuzzle Team <support@kuzzle.io>",
-  "version": "3.0.14",
```
Comment (Aschen) [inline package.json]: You only have to update the version if this is a hotfix, but the destination branch is `3-dev`

Comment (Yoann-Abbes) [reply package.json]: Hmm it's updated automatically

File: lib/errors/kuzzleError.js [hunk 3]
```diff
@@ -7,13 +7,14 @@ const util = require('util');
  */
 class KuzzleError extends Error {
 
-  constructor(message, status, errorName = 'Undocumented error', code = 0) {
+  constructor(message, status, errorName = 'internal.errorsManager.undocumented_error', code = 0) {
```
Comment (scottinet) [inline lib/errors/kuzzleError.js]: so... what's the point in concatenating these information in kuzzle if we have to split them afterward in the kuzzleError object?

Comment (Aschen) [reply lib/errors/kuzzleError.js]: image

**Pull request closed without merging**